catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/XstreamPathConverter.java

```diff
@@ -164,13 +166,11 @@ protected boolean doBasicPathsMatch(final Path pathObj1, final Path pathObj2) {
      */
     private boolean endsMatch(char[] path1, int index1, char[] path2, int index2) {
```

Comment:
I'm wondering if we should take a different approach on this. Instead of this complex logic to walk both strings for comparison, we could have simpler code that efficiently strips out the xpath index characters and the xpath attribute indicator. Then just compare those strings.